src/render/filters.rs

```diff
@@ -299,6 +360,71 @@ mod tests {
         })
     }
 
+    #[test]
+    fn test_render_filter_slugify() {
+        pyo3::prepare_freethreaded_python();
+
+        Python::with_gil(|py| {
+            let engine = EngineData::empty();
+            let template_string = "{{ var|slugify }}".to_string();
+            let context = PyDict::new(py);
+            context.set_item("var", "hello world").unwrap();
+            let template = Template::new_from_string(py, template_string, &engine).unwrap();
+            let result = template.render(py, Some(context), None).unwrap();
+
+            assert_eq!(result, "hello-world");
+
+            let engine = EngineData::empty();
+            let template_string = "{{ var|slugify }}".to_string();
+            let context = PyDict::new(py);
+            context.set_item("var", " hello world").unwrap();
+            let template = Template::new_from_string(py, template_string, &engine).unwrap();
+            let result = template.render(py, Some(context), None).unwrap();
+
+            assert_eq!(result, "hello-world");
```

Comment:
Splitting the tests up would also allow the test name to point to the difference between these two tests - the leading whitespace in the second.

Reply:
Agreed, collecting them in the same test was partially laziness, partially keeping them a bit more organized compared to have single test functions. 

All the tests have been broken up now though.